src/zenml/exceptions.py

```diff
@@ -212,6 +212,11 @@ class SecretExistsError(ZenMLBaseException):
     already exists."""
 
 
+class SecretDoesNotExistError(ZenMLBaseException):
```

Comment:
I think we should remove this error and replace all occurrences with `KeyError` instead. Those `KeyErrors` are also raised in the other secrets managers, and having a common exception for all of those allows us to write
```python
try:
    secrets_manager.get_secret(...)
except KeyError:
    # do something to handle the case that the secret doesn't exist
```
in other places in the code.